src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -96,12 +113,16 @@ private void SetFrequency(int frequency)
             {
                 throw new ArgumentOutOfRangeException(nameof(frequency), frequency, "Value must not be negative.");
             }
-
+            double oldDutyCycle = _dutyCycle;
```

Comment:
This specific fix is already included in https://github.com/dotnet/iot/pull/758 and there are some edge cases around it which I already fixed.

There is couple of more issues with PWM in general which I'm still investigating (some of them make the driver not usable until RPi is rebooted).

Reply:
@krwq Oh... I didn't see that.
The only strange issue I had once was that the PWM channel would behave as if I had set the frequency to 1Hz. Couldn't reproduce this after a reboot. Maybe we need support for register based PWM access anyway, if this is so instable?

Any idea when that PR is ready for merge?

Reply:
@pgrawehr cannot merge until I upgrade the lab, currently we are waiting for PCBs to arrive.

I'll isolate that change from my PR so that we can merge this sooner but I'd prefer to do more testing first since I really want to figure out that issue which causes PWM to be unusable until reboot (currently I'm a bit busy with some non IoT related stuff)

Reply:
That shipment is seemingly taking ages, then... 
I have no hurry getting this to work, as I probably won't need it right now. I also just got a PCA9685 delivered, which is probably the better fit for running a number of (RGB) Leds. 
@krwq Question on a side issue: The library currently doesn't support setting the GPIO Pins to something else than IN or OUT. Namely, it is not possible to set any pin to their ALTn modes after they have been set to in or out. Are there already plans to implement this somehow? I think this is required as part of bringing hardware into the CI process, otherwise any wrong pin setup in a build would require a reboot afterwards.

Reply:
Yes, I agree that shipment is kinda slow (it's second version of the prototype though). @pgrawehr can you give a specific example of the scenario which requires reboot? (please file a separate issue so it doesn't get lost)

Reply:
Created #857 on this (sorry for the lengthy explanation)